src/lcm/argmax.py

```diff
@@ -1,31 +1,45 @@
 import jax
 import jax.numpy as jnp
+from jax import Array
+from jax.typing import ArrayLike
 
 # ======================================================================================
 # argmax
 # ======================================================================================
 
 
-def argmax(a, axis=None, initial=None, where=None):
-    """Compute the argmax of a n-dim array along axis.
+def argmax(
+    a: ArrayLike,
+    axis: int | tuple[int, ...] | None = None,
+    initial: ArrayLike | None = None,
+    where: ArrayLike | None = None,
+) -> tuple[Array, Array]:
+    """Compute the argmax of an n-dim array along axis.
+
+    If multiple maxima exist, the first index will be selected.
 
     Args:
-        a (jax.numpy.ndarray): Multidimensional jax array.
+        a (ArrayLike): Multidimensional array.
         axis (int | tuple | None): Axis along which to compute the argmax. If None, the
             argmax is computed over all axes.
-        initial (scalar): The minimum value of an output element. Must be present to
+        initial (ArrayLike): The minimum value of an output element. Must be present to
             allow computation on empty slice. See ~numpy.ufunc.reduce for details.
-        where (jax.numpy.ndarray): Boolean array of the same shape as a. Only the values
-            for which where is True are considered for the argmax.
+        where (ArrayLike): Elements to compare for the maximum. See ~numpy.ufunc.reduce
+            for details.
 
     Returns:
-        - jax.numpy.ndarray: Array with the same shape as a, except for the dimensions
-          specified in axis, which are dropped. The value corresponds to an index
-          that can be translated into a tuple of indices using jnp.unravel_index.
+        - Array: The argmax indices. Array with the same shape as a, except for
+            the dimensions specified in axis, which are dropped. The value corresponds
+            to an index that can be translated into a tuple of indices using
+            jnp.unravel_index.
+
+        - Array: The corresponding maximum values.
 
     """
     # Preparation
     # ==================================================================================
+    a = jnp.asarray(a)
```

Comment:
I am sure you do this and the typing to ArrayLike instead of Array for a reason, but my naive mind thought we'd always have jnp.arrays only. Maybe add a hint somewhere why we can't assume that?

Reply:
> I am sure you do this and the typing to ArrayLike instead of Array for a reason

This does not mean that the reason is a good one!

After reading a lot about best practices with JAX typing, I got the impression that a standard approach is weak input types and strict output types (see [here](https://jax.readthedocs.io/en/latest/jax.typing.html#jax-typing-best-practices)).

However, as you are implicitly pointing out if we never use the weaker input types internally, then there is also no reason at all not to use the stricter input type.

I've updated this, thank you!